src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -880,13 +573,676 @@ def defineFaceAnnotations(cls, region, options, annotationGroups):
+            xAround.append(x)
+            d2Around.append(d2)
+            # Calculate d1
+            if n2 > 0:
+                v1 = xRaw[n1][n2]
+                v2 = xRaw[n1 + 1 if n1 < elementsCountAround - 2 else 0][n2]
+                d1 = findDerivativeBetweenPoints(v1, v2)
+                d1Around.append(d1)
+            else:
+                d1Around.append(d2Raw[n2][0])
+        if n2 > 0:
+            d1Smoothed = interp.smoothCubicHermiteDerivativesLoop(xAround, d1Around)
+        else:
+            d1Smoothed = d1Around
+        xSampledDome.append(xAround)
+        d1SampledDome.append(d1Smoothed)
+        d2SampledDome.append(d2Around)
+
+    return xSampledDome, d1SampledDome, d2SampledDome
+
+def findNodesAlongBladderNeck(sx_dome_group, sx_neck_group, domeSegmentLength, neckSegmentLength, elementsCountAround, elementsCountAlongNeck):
+
+    # Transition
+    transitLength = (domeSegmentLength + neckSegmentLength) / 2
+    addingLength = transitLength - neckSegmentLength
+    if transitLength <= neckSegmentLength:
+        e = 1
+        xi = transitLength / neckSegmentLength
+    else:
+        e = int(transitLength / neckSegmentLength) + 1
+        xi = transitLength / (e * neckSegmentLength)
+    xTransition = interp.interpolateCubicHermite(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    # d1Transition = interp.interpolateCubicHermiteDerivative(sx_dome_group[0][-1], sx_dome_group[1][-1], sx_neck_group[0][e], sx_neck_group[1][e], xi)
+    d2Transition = interp.interpolateCubicHermite(sx_dome_group[2][-1], sx_dome_group[3][-1], sx_neck_group[2][e], sx_neck_group[3][e], xi)
+    d3Transition = interp.interpolateCubicHermite(sx_dome_group[4][-1], sx_dome_group[5][-1], sx_neck_group[4][e], sx_neck_group[5][e], xi)
+    px_transit, pd1_transit = createEllipsePoints(xTransition, 2 * math.pi, d2Transition, d3Transition, elementsCountAround,
+                                                  startRadians=0.0)
+
+    # Create ellipses along neck around the central path
+    xEllipses_neck = []
+    d1Ellipses_neck = []
+    for n in range(0, len(sx_neck_group[0])):
+        px, pd1 = createEllipsePoints(sx_neck_group[0][n], 2 * math.pi, sx_neck_group[2][n], sx_neck_group[4][n], elementsCountAround,
+                                      startRadians=0.0)
+        xEllipses_neck.append(px)
+        d1Ellipses_neck.append(pd1)
+
+    # Find d2
+    d2Raw = []
+    for n1 in range(elementsCountAround):
+        xAlong = []
+        d2Along = []
+        for n2 in range(len(xEllipses_neck) - 1):
+            v1 = xEllipses_neck[n2][n1]
+            v2 = xEllipses_neck[n2 + 1][n1]
+            d2 = findDerivativeBetweenPoints(v1, v2)
+            xAlong.append(v1)
+            d2Along.append(d2)
+        xAlong.append(xEllipses_neck[-1][n1])
+        d2Along.append(d2)
+        d2Smoothed = interp.smoothCubicHermiteDerivativesLine(xAlong, d2Along)
+        d2Raw.append(d2Along)
```

Comment:
Start d2 should be known from the Body nodes around the junction ellipse and their directions should be used here. Likewise d2 a the urethra should be prescribed from the central path node (and all the central paths supplied should have D12 and D13 zeroed) -- use the algorithm I suggested for the Dome-Neck junction earlier to handle the full case even for zero D12 and D13.

Reply:
Added the start d2 from the junction ellipse.